applications/StructuralMechanicsApplication/custom_processes/distribute_load_on_surface_process.cpp

```diff
@@ -0,0 +1,68 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Armin Geiser
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "distribute_load_on_surface_process.h"
+#include "utilities/interval_utility.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+DistributeLoadOnSurfaceProcess::DistributeLoadOnSurfaceProcess(ModelPart& rModelPart,
+                                                                Parameters Settings)
+                                                                : mrModelPart(rModelPart),
+                                                                mParameters(Settings)
+{
+    Parameters default_parameters(R"(
+        {
+            "help"            : "This process distributes a load on surface load conditions belonging to a modelpart. The load is distributed according to the surface area.",
+            "model_part_name" : "please_specify_model_part_name",
+            "interval"        : [0.0, 1e30],
+            "load"           : [1.0, 0.0, 0.0]
+        }  )"
+    );
+
+    IntervalUtility interval_utility(mParameters);
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+    KRATOS_ERROR_IF(mParameters["load"].GetVector().size() != 3) <<
+        "'load' has to be a vector of doubles with size 3!" << std::endl;
+}
+
+void DistributeLoadOnSurfaceProcess::ExecuteInitializeSolutionStep()
+{
+    const double current_time = mrModelPart.GetProcessInfo().GetValue(TIME);
```

Comment:
```suggestion
    const double current_time = mrModelPart.GetProcessInfo()[TIME];
```

usually done like this